File: AirTECWebAPI/Models/Execution.cs
using System;
using System.Collections.Generic;

namespace AirTECWebAPI.Models;

public partial class Execution
{
    public int Idexecution { get; set; }

    public int? NumberFlight { get; set; }

    public string? PlateNumber { get; set; }

    public DateOnly? Date { get; set; }

    public TimeOnly? DepartureTime { get; set; }

    public int? Price { get; set; }

    public string? Status { get; set; }

    public int? BoardingDoor { get; set; }

    public virtual Flight? NumberFlightNavigation { get; set; }

    public virtual Plane? PlateNumberNavigation { get; set; }

    public virtual ICollection<Promotion> Promotions { get; set; } = new List<Promotion>();

    public virtual ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();

    public virtual ICollection<Seat> Seats { get; set; } = new List<Seat>();

    public virtual ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
}

File: frontend/src/app/Interfaces/execution.ts
type TimeOnly = string;
//revisar el tipo de TimeOnly
export class Execution {
    idexecution?: number;
    numberFlight?: number;
    plateNumber?: string;
    date?: Date;
    departureTime?: TimeOnly;
    price?: number;
    status?: string;
    boardingDoor?: number;
}

export class Promotion {
    number?: number;
    idexecution?: number;
    period?: string;
    price?: number;
}

export class Reservation {
    idreservation?: number;
    iduser?: number;
    idexecution?: number;
}

export class Seat {
    seatNumber?: number;
    idexecution?: number;
}

export class Ticket {
    idticket?: number;
    taxes?: number;
    totalAmount?: number;
    seatNumber?: number;
    idexecution?: number;
    idpassenger?: number;
    iduser?: number;
}
